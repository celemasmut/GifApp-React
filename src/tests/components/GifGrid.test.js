import { shallow } from 'enzyme'
import React from 'react';
import '@testing-library/jest-dom'
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')  // se finje su llamada a ese archivo


describe('Prueba en GifGrid', () => {

    const category = "mi categoria";
    
    
    test('debe mostrarse correctamente', () => {
        

        useFetchGifs.mockReturnValue( {
            data: [],
            loading:true
        });
        const wrapper = shallow( <GifGrid category= {category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se carga imagenes con useFetchGifs', () => {
        
        const gifs = [{
            id:'DFG',
            url: 'https://localhost/cualka/cuak.jpg',
            title:'cualka'
        }]

        useFetchGifs.mockReturnValue( {
            data: gifs,
            loading:false
        });

        const wrapper = shallow( <GifGrid category= {category} />);

        expect(wrapper).toMatchSnapshot();

        expect( wrapper.find('p').exists()).toBe(false);

        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);

    });
    
    
    
})
