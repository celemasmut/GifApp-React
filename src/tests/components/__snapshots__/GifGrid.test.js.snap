// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en GifGrid debe de mostrar items cuando se carga imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__headShake"
  >
    mi categoria
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="DFG"
      key="DFG"
      title="cualka"
      url="https://localhost/cualka/cuak.jpg"
    />
  </div>
</Fragment>
`;

exports[`Prueba en GifGrid debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__headShake"
  >
    mi categoria
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
